rest/src/server/bootstrapper.js

```diff
@@ -93,23 +94,59 @@ module.exports = {
 
 	/**
 	 * Creates a REST api server.
-	 * @param {array} crossDomainConfig Configuration related to access control, contains allowed host and HTTP methods.
+	 * @param {array} config Application configuration
 	 * @param {object} formatters Formatters to use for formatting responses.
 	 * @param {object} throttlingConfig Throttling configuration parameters, if not provided throttling won't be enabled.
 	 * @returns {object} Server.
 	 */
-	createServer: (crossDomainConfig, formatters, throttlingConfig) => {
-		// create the server using a custom formatter
-		const server = restify.createServer({
+	createServer: (config, formatters, throttlingConfig) => {
+		let serverOptions = {
 			name: '', // disable server header in response
 			formatters: {
 				'application/json': formatters.json
 			}
-		});
+		};
+		if (!config.protocol)
+			winston.warn('Protocol is not configured explicitly in the configuration, defaulting to HTTPS.');
```

Comment:
isn't the default in the config HTTPS? so, the user would be explicitly turning it off by leaving it empty?

in any event, it might be better to make this a noisy error (i.e. fail to start) instead of allowing a potential misconfiguration.

Reply:
Users can only turn it off by setting it to HTTP, leaving empty will default to the HTTPS and in that case, the REST server won't start since it'll be asking for the SSL key and certificate, so yes, it'll be a noisy error.